cmd/cluster-init/update_bootstrap_secrets.go

```diff
@@ -0,0 +1,348 @@
+				RegistryURL: fmt.Sprintf("registry.%s.ci.openshift.org", clusterName),
+			},
+		},
+	}
+	itemContext.DockerConfigJSONData =
+		append(itemContext.DockerConfigJSONData, items...)
+	return itemContext
+}
+
+func registryPullTokenField(clusterName string) string {
+	return fmt.Sprintf("token_image-puller_%s_reg_auth_value.txt", clusterName)
+}
+
+func generateDockerConfigJsonSecretConfigTo(name string, namespace string, clusterName string) secretbootstrap.SecretContext {
+	return secretbootstrap.SecretContext{
+		Cluster:   clusterName,
+		Name:      name,
+		Namespace: namespace,
+		Type:      "kubernetes.io/dockerconfigjson",
+	}
+}
+
+func updatePodScalerSecret(c *secretbootstrap.Config, o options) error {
+	key := fmt.Sprintf("%s.%s", o.clusterName, config)
+	return appendSecretItemContext(c, podScaler, string(api.ClusterAPPCI), key, secretbootstrap.ItemContext{
+		Field: serviceAccountKubeconfigPath(podScaler, o.clusterName),
+		Item:  podScaler,
+	})
+}
+
+func updateDPTPControllerManagerSecret(c *secretbootstrap.Config, o options) error {
+	const DPTPControllerManager = "dptp-controller-manager"
+	keyAndField := serviceAccountKubeconfigPath(DPTPControllerManager, o.clusterName)
+	return appendSecretItemContext(c, DPTPControllerManager, string(api.ClusterAPPCI), keyAndField, secretbootstrap.ItemContext{
+		Field: keyAndField,
+		Item:  buildUFarm,
+	})
+}
+
+func updateRehearseSecret(c *secretbootstrap.Config, o options) error {
+	keyAndField := serviceAccountKubeconfigPath(ciOperator, o.clusterName)
+	return appendSecretItemContext(c, "pj-rehearse", string(api.ClusterBuild01), keyAndField, secretbootstrap.ItemContext{
+		Field: keyAndField,
+		Item:  buildUFarm,
+	})
+}
+
+func updateChatBotSecret(c *secretbootstrap.Config, o options) error {
+	const chatBot = "ci-chat-bot"
+	name := chatBot + "-kubeconfigs"
+	keyAndField := serviceAccountKubeconfigPath(chatBot, o.clusterName)
+	return appendSecretItemContext(c, name, string(api.ClusterAPPCI), keyAndField, secretbootstrap.ItemContext{
+		Field: keyAndField,
+		Item:  chatBot,
+	})
+}
+
+func appendSecretItemContext(c *secretbootstrap.Config, name string, cluster string, key string, value secretbootstrap.ItemContext) error {
+	logrus.Infof("Appending secret item to: {name: %s, cluster: %s}", name, cluster)
+	sc, err := findSecretConfig(name, cluster, c.Secrets)
+	if err != nil {
+		return err
+	}
+	sc.From[key] = value
+	return nil
+}
+
+func updateExistingRegistryPullCredentialsAllSecrets(c *secretbootstrap.Config, o options) error {
+	for _, cluster := range c.UserSecretsTargetClusters {
+		if cluster != string(api.ClusterHive) && cluster != arm01 && cluster != o.clusterName {
+			return appendRegistrySecretItemContext(c, regPullCredsAll, cluster, secretbootstrap.DockerConfigJSONData{
+				AuthField:   fmt.Sprintf("token_image-puller_%s_reg_auth_value.txt", o.clusterName),
+				Item:        buildUFarm,
+				RegistryURL: fmt.Sprintf("registry.%s.ci.openshift.org", o.clusterName),
```

Comment:
This could use a function in the same files that declare the central registry - registry domain for cluster name

Reply:
There actually already is a `func RegistryDomainForClusterName(ClusterName string) (string, error) {` in `api/domain.go`. It intentionally errors out with any cluster other than `app.ci`, but it is currently completely unused (other than the unit test for it). So I will repurpose this function.